lib/iaas/BackupStore.js

```diff
@@ -291,7 +291,10 @@ class Filename {
       .zipObject(path.basename(filename).split('.'))
       .assign(data)
       .tap(filenameObject => {
-        if (filenameObject.started_at) {
+        if (this.keys.exclude_file_with_root_value !== undefined && this.keys.exclude_file_with_root_value === _.get(filenameObject, this.root)) {
```

Comment:
zipObject would fail if `path.basename(filename).split('.).length !== this.keys.length` so may be this could be used as a condition rather than having this check.

Also, if such a check is done then OOB backups will be excluded from being cleaned by BackupReaper. Backupreaper must clean up both OOB and instances scheduled backups!.

Reply:
when you say , zipObject would fail , you mean some of the key's will have value as `undefined` correct ? It would not error out as seen now . 

`path.basename(filename).split('.).length !== this.keys.length` - this does not guarantee that we will filter out OOB backups right ? Since in the future if the keys in BackupStoreForOOB = keys in BackupStoreForServiceInstance , then this check will be in-affective ?!

> Also, if such a check is done then OOB backups will be excluded from being cleaned by BackupReaper. Backupreaper must clean up both OOB and instances scheduled backups!.

Backup reaper job even now doesnt process OOB backups , correct ? Will check this,

Reply:
Created an issue for handling backup reaper job use case for OOB backups - https://github.com/cloudfoundry-incubator/service-fabrik-broker/issues/190 . Will send it in another PR.

Reply:
ok